### DISCUSSION TITLE
Two questions: Is there a start_date param for any endpoint? How to account for dividend price changes?

### DISCUSSION THREAD

**Discussion by melgazar9** [ranaroussi/yfinance#1864] [Q&A]
Hello I have two questions about the yfinance library.

1. Is there a general parameter passed to kwargs that I can set to get the start_date of any endpoint? For example we can set `start_date` and `end_date` to the `history()` method but can we set these to any method (e.g. `institutional_holders`, `insider_purchases`, `upgrades_downgrades`, etc)?
2. For the `history()` method, let's say I have a job that pulls GOOG daily data once per day with the start date set to the last date it ran. Say the job runs on 2/1/2020 --> then the next job runs on 2/2/2020 with the start_date set to 2/1. If there's a stock split on 2/2/2020 is there a way to have it automatically account for that? I was thinking by using the built-in cacheing maybe it can recognize splits in some way? If not it probably just makes sense to keep track of the splits in a database and apply basic logic to the tickers that had a split.

Thanks!

**Comment by ValueRaider** [comment]
### `history` caching

Check out https://github.com/ValueRaider/yfinance-cache
